Upstream: https://bugzilla.gnome.org/show_bug.cgi?id=727440

From 2c493bbe144a7791b3a78e3428b713f6eab69aa9 Mon Sep 17 00:00:00 2001
From: Calvin Walton <calvin.walton@kepstin.ca>
Date: Tue, 1 Apr 2014 10:39:54 -0400
Subject: [PATCH] Add configure option to enable/disable wayland support.

Right now, wayland support is autodetected based on the presence of the
mutter-wayland package on the system, which makes life difficult for
packagers.

Add an option to the configure script that allows deterministicly
enabling/disabling wayland support.
---
 configure.ac | 27 +++++++++++++++++++++------
 1 file changed, 21 insertions(+), 6 deletions(-)

diff --git a/configure.ac b/configure.ac
index 9a30e65..ed67ad0 100644
--- a/configure.ac
+++ b/configure.ac
@@ -112,13 +112,27 @@ fi
 
 PKG_CHECK_MODULES(GNOME_SHELL, $SHARED_PCS)
 PKG_CHECK_MODULES(MUTTER, libmutter >= $MUTTER_MIN_VERSION)
-PKG_CHECK_MODULES(MUTTER_WAYLAND, [libmutter-wayland >= $MUTTER_MIN_VERSION],
-                 [MUTTER_WAYLAND_TYPELIB_DIR=`$PKG_CONFIG --variable=typelibdir libmutter-wayland`
-                  AC_SUBST(MUTTER_WAYLAND_TYPELIB_DIR)
-                  have_mutter_wayland=yes],
-                 [have_mutter_wayland=no])
 
-AM_CONDITIONAL(HAVE_MUTTER_WAYLAND, test $have_mutter_wayland != no)
+AC_ARG_ENABLE(wayland,
+              AS_HELP_STRING([--disable-wayland],
+                            [disable Wayland compositor support  @<:@default=auto@:>@]),,
+              [enable_wayland=auto])
+
+if test "x$enable_wayland" != "xno"; then
+   PKG_CHECK_MODULES(MUTTER_WAYLAND, [libmutter-wayland >= $MUTTER_MIN_VERSION],
+                    [MUTTER_WAYLAND_TYPELIB_DIR=`$PKG_CONFIG --variable=typelibdir libmutter-wayland`
+                     AC_SUBST(MUTTER_WAYLAND_TYPELIB_DIR)
+                     have_mutter_wayland=yes],
+                    [have_mutter_wayland=no])
+else
+   have_mutter_wayland="no  (disabled)"
+fi
+
+if test "x$have_mutter_wayland" != "xyes" && test "x$enable_wayland" = "xyes"; then
+   AC_MSG_ERROR([Couldn't find mutter-wayland.])
+fi
+
+AM_CONDITIONAL(HAVE_MUTTER_WAYLAND, test $have_mutter_wayland = yes)
 
 PKG_CHECK_MODULES(GNOME_SHELL_JS, gio-2.0 gjs-internals-1.0 >= $GJS_MIN_VERSION)
 PKG_CHECK_MODULES(ST, clutter-1.0 gtk+-3.0 libcroco-0.6 >= 0.6.8 x11)
@@ -264,4 +278,5 @@ Build configuration:
 
        Support for NetworkManager:             $have_networkmanager
        Support for GStreamer recording:        $build_recorder
+       Support for Wayland:                    $have_mutter_wayland
 "
-- 
1.9.1

